NitroxServer/GameLogic/Entities/EntityData.cs

```diff
@@ -164,12 +192,53 @@ public Optional<Entity> GetEntityByGuid(string guid)
         {
             Entity entity = null;
 
+            TryGetParentOrChild(guid, out entity);
+
+            return Optional<Entity>.OfNullable(entity);
+        }
+
+        private bool TryGetParentOrChild(string guid, out Entity entity)
```

Comment:
It looks like Get/Remove have turned into very expensive and complex operations.  It is possible to keep entitiesByGuid a flat map of all entities (including children) then only return items that are not children in the phasing logic above?  That would allow all of these methods to remain as-is with only a small change above.